src/arithmetic_impls.rs

```diff
@@ -0,0 +1,255 @@
+    }
+}
+
+impl CheckedRem for Decimal {
+    #[inline]
+    fn checked_rem(&self, v: &Decimal) -> Option<Decimal> {
+        Decimal::checked_rem(*self, *v)
+    }
+}
+
+forward_all_binop!(impl Div for Decimal, div);
+impl<'a, 'b> Div<&'b Decimal> for &'a Decimal {
+    type Output = Decimal;
+
+    #[inline]
+    fn div(self, other: &Decimal) -> Decimal {
+        match ops::div_impl(self, other) {
+            CalculationResult::Ok(quot) => quot,
+            CalculationResult::Overflow => panic!("Division overflowed"),
+            CalculationResult::DivByZero => panic!("Division by zero"),
+        }
+    }
+}
+
+forward_all_binop!(impl Mul for Decimal, mul);
+impl<'a, 'b> Mul<&'b Decimal> for &'a Decimal {
+    type Output = Decimal;
+
+    #[inline]
+    fn mul(self, other: &Decimal) -> Decimal {
+        match ops::mul_impl(self, other) {
+            CalculationResult::Ok(prod) => prod,
+            _ => panic!("Multiplication overflowed"),
+        }
+    }
+}
+
+forward_all_binop!(impl Rem for Decimal, rem);
+impl<'a, 'b> Rem<&'b Decimal> for &'a Decimal {
+    type Output = Decimal;
+
+    #[inline]
+    fn rem(self, other: &Decimal) -> Decimal {
+        match ops::rem_impl(self, other) {
+            CalculationResult::Ok(rem) => rem,
+            CalculationResult::Overflow => panic!("Division overflowed"),
+            CalculationResult::DivByZero => panic!("Division by zero"),
+        }
+    }
+}
+
+forward_all_binop!(impl Sub for Decimal, sub);
+impl<'a, 'b> Sub<&'b Decimal> for &'a Decimal {
+    type Output = Decimal;
+
+    #[inline(always)]
+    fn sub(self, other: &Decimal) -> Decimal {
+        match ops::sub_impl(self, other) {
+            CalculationResult::Ok(sum) => sum,
+            _ => panic!("Subtraction overflowed"),
+        }
+    }
+}
+
+#[cfg(test)]
+mod tests {
+    use crate::Decimal;
+
+    #[test]
+    fn saturated_methods_have_correct_output() {
+        assert_eq!(Decimal::MAX.checked_add(Decimal::MAX), None);
+        assert_eq!(Decimal::MAX.checked_div(Decimal::ZERO), None);
+        assert_eq!(Decimal::MAX.checked_mul(Decimal::MAX), None);
+        assert_eq!(Decimal::MAX.checked_rem(Decimal::ZERO), None);
+        assert_eq!(Decimal::MAX.checked_sub(Decimal::MIN), None);
+
+        assert_eq!(Decimal::MAX.saturating_add(Decimal::MAX), Decimal::MAX);
+        assert_eq!(Decimal::MAX.saturating_mul(Decimal::MAX), Decimal::MAX);
+        assert_eq!(Decimal::MAX.saturating_sub(Decimal::MIN), Decimal::MAX);
+
+        assert_eq!(Decimal::MIN.checked_add(Decimal::MIN), None);
+        assert_eq!(Decimal::MIN.checked_div(Decimal::ZERO), None);
+        assert_eq!(Decimal::MIN.checked_mul(Decimal::MIN), None);
+        assert_eq!(Decimal::MIN.checked_rem(Decimal::ZERO), None);
+        assert_eq!(Decimal::MIN.checked_sub(Decimal::MAX), None);
+
+        assert_eq!(Decimal::MIN.saturating_add(Decimal::MIN), Decimal::MIN);
+        assert_eq!(Decimal::MIN.saturating_mul(Decimal::MIN), Decimal::MIN);
```

Comment:
As above - I think this should actually result in `MAX` since two negatives multiplied would be a large number. Multiplication should perhaps have some sign combinations just to make sure it's handled correctly.